cardano-crypto-tests/src/Test/Crypto/EllipticCurve.hs

```diff
@@ -146,6 +162,16 @@ testCommut :: (Show a, Eq a) => (a -> a -> a) -> a -> a -> Property
 testCommut f a b =
   f a b === f b a
 
+prop_repeatedAddition :: forall curve. BLS.BLS curve => Integer -> BLS.Point curve -> Property
+prop_repeatedAddition a p = BLS.blsMult p red_a === repeatedAdd red_a p
+    where
+    red_a = mod a 1000
+    repeatedAdd :: Integer -> BLS.Point curve -> BLS.Point curve
+    repeatedAdd scalar point =
+         if scalar >= 0
+         then foldl' BLS.blsAddOrDouble BLS.blsZero $ genericReplicate scalar point
+         else repeatedAdd (-scalar) (BLS.blsNeg point)
```

Comment:
There is no danger of large values when generating `Integer` or `Int` with QuickChecks arbitrary instance, so `mod` is totally redundant. Also a bit more simplification, `generic*` aren't commonly used functions, relying on `Int` is better, afterall we aren't planning on generating lists with more than 9223372036854775807 values :slightly_smiling_face: 
```suggestion
prop_repeatedAddition :: forall curve. BLS.BLS curve => Int -> BLS.Point curve -> Property
prop_repeatedAddition a p = BLS.blsMult p (fromIntegral a) === repeatedAdd a p
    where
    repeatedAdd :: Int -> BLS.Point curve -> BLS.Point curve
    repeatedAdd scalar point =
         foldl' BLS.blsAddOrDouble BLS.blsZero $ replicate (abs scalar) (BLS.blsCneg point (scalar < 0))
```